### PR TITLE
Twig 3 support

### PR DESCRIPTION
**Pull request by xvilo**
- Add twig 3 as a dependency
- Update the stub environment to reflect class changes
- Fixes Code style issues that
- Fixes PHP Unit warnings, so all green now :) 
- Change php required version 7.2 (as 7.1 is not supported by twig anymore)
    - Update the travis file to include 7.2 -> 7.4 versions which are currently maintained 
- Should probably be a new major version (2.0.0)

### CODE REVIEW

**Comment by xvilo** [comment]
Tests are now passing now, fixed some stuff and added Twig 3 as dep. Can this be reviewed @sserbin?

**Comment by xvilo** [comment]
Hi @sserbin, when do you expect to take a look at this? We would love to start using twig 3 with your linter :)

File: src/StubEnvironment.php [hunk 1]
```diff
@@ -13,7 +14,7 @@ class StubEnvironment extends Environment
     /**
      * {@inheritdoc}
      */
-    public function getFilter($name)
+    public function getFilter(string $name): ?TwigFilter
```
Comment (sserbin) [inline src/StubEnvironment.php]: if I'm not mistaken, a signature like below would be compatible with both 2.x and 3.x of twig
```php
public function getFilter($name): ?TwigFilter
```

File: composer.json [hunk 2]
```diff
@@ -16,10 +16,10 @@
         "psr-4": { "Sserbin\\TwigLinter\\Tests\\": "tests/"}
     },
     "require": {
-        "php": "~7.1",
+        "php": "~7.2",
         "symfony/console": "^4.1",
         "symfony/finder": "^4.1",
-        "twig/twig": "^2.5",
+        "twig/twig": "3",
```
Comment (sserbin) [inline composer.json]: this could be made compatible for both with `^2.5 || ^3`

File: src/StubEnvironment.php (hunk 1 above)
Comment (xvilo) [reply src/StubEnvironment.php]: Yes, this worked for all but `getTest`, I have added an if statement with a comment to clarify.

**Comment by xvilo** [comment]
@sserbin I have added twig 2 compatibilities again
1. update composer file
2. changed inline types to only include return types
3. added a small if statement for type check to make it work on both (latest minor) versions.

See:

### Twig 2
Screenshot 2019-12-23 at 23 26 54

### Twig 3
Screenshot 2019-12-23 at 23 25 59

**Comment by xvilo** [comment]
@sserbin what to do with the linting issues?

**Comment by sserbin** [comment]
> @sserbin what to do with the linting issues?

fixed those ;)

**Pull request merged**